R/capture-condition.R

```diff
@@ -1,3 +1,25 @@
+
+bang <- function(expr, env = caller_env()) {
+  eval_bare(enexpr(expr), env)
+}
+new_capture <- function(class) {
+  exiting_handlers <- rep_named(class, list(identity))
+
+  calling_handlers <- rep_named(class, alist(function(cnd) {
+    cnd <- cnd_entrace(cnd)
+    return_from(env, cnd)
+  }))
+
+  bang(function(code, entrace = FALSE) {
+    if (!entrace) {
+      return(tryCatch({ code; NULL }, !!!exiting_handlers))
+    }
+
+    env <- environment()
```

Comment:
I think this would be simpler to follow if you required `class` to be length one so you could use `!!class := function(cnd) ... `

Reply:
We need to allow deep name-unquoting in rlang first.

Reply:
Ooooh, bummer.